src/components/tx-flow/flows/RecoveryAttempt/RecoveryAttemptReview.tsx

```diff
@@ -0,0 +1,93 @@
+import { type SyntheticEvent, useState, useContext, useCallback } from 'react'
+import { CircularProgress, CardActions, Button, Typography, Stack, Divider } from '@mui/material'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import CheckWallet from '@/components/common/CheckWallet'
+import { Errors, trackError } from '@/services/exceptions'
+import { dispatchRecoveryExecution } from '@/features/recovery/services/recovery-sender'
+import useWallet from '@/hooks/wallets/useWallet'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import type { TransactionAddedEvent } from '@gnosis.pm/zodiac/dist/cjs/types/Delay'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import FieldsGrid from '@/components/tx/FieldsGrid'
+import { TxModalContext } from '../..'
+
+export type RecoveryAttemptReviewProps = {
+  params: {
+    args: TransactionAddedEvent.Log['args']
+    address: string
+  }
+}
+
+const RecoveryAttemptReview = ({ params }: RecoveryAttemptReviewProps) => {
+  const [isPending, setIsPending] = useState(false)
+  const [error, setError] = useState<Error>()
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const { setTxFlow } = useContext(TxModalContext)
+
+  const onFormSubmit = useCallback(
```

Comment:
I think we do not need to use a `useCallback` here, because this function doesn't do expansive calculation and also it is only used as a DOM event.

Reply:
It's not about whether the function itself is heavy or not. It's not going to be called on render anyway, only created. We memoize callbacks because otherwise it will trigger a re-rendering of the JSX tree as the onSubmit prop would always receive a new reference. While it might be only a slight optimization in this case, it doesn't hurt.